source/sso_token_utils.c

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/auth/private/sso_token_utils.h>
+#include <aws/cal/hash.h>
+#include <aws/common/encoding.h>
+#include <aws/common/file.h>
+#include <aws/common/json.h>
+
+#if defined(_MSC_VER)
+#    pragma warning(disable : 4232)
+#endif /* _MSC_VER */
+
+AWS_STATIC_STRING_FROM_LITERAL(s_sso_cache_directory, "/.aws/sso/cache/");
+
+struct aws_string *aws_construct_token_path(struct aws_allocator *allocator, const struct aws_string *input) {
```

Comment:
I have renamed it to `aws_construct_sso_token_path`. In general, `token` is a generic term which is just a string. It can be anything like accessToken, sessionToken, bearer token etc. `sso_token` is a json used by the sso service, which contains the following 
```
{
    "accessToken": " ",
    "expiresAt": "",
    "refreshToken": "string",
    "clientId": " ",
    "clientSecret": " ",
    "registrationExpiresAt": " ",
    "region": " ",
    "startUrl": " "
}
```
Currently, we are only using accessToken out of these fields thus they look similar, but when we implement the refresh functionality, we will need other things like refresh token.